src/nupic/algorithms/Connections.hpp

```diff
@@ -41,15 +41,15 @@ namespace nupic
 
     namespace connections
     {
-      typedef UInt32 CellIdx;
-      typedef unsigned char SegmentIdx;
-      typedef unsigned char SynapseIdx;
+      typedef UInt CellIdx;
+      typedef short SegmentIdx;
+      typedef unsigned short SynapseIdx;
       typedef Real32 Permanence;
       typedef UInt64 Iteration;
 
-      #define CELL_MAX (USHRT_MAX-1)
-      #define SEGMENT_MAX (UCHAR_MAX-1)
-      #define SYNAPSE_MAX (UCHAR_MAX-1)
+      #define CELL_MAX ((USHRT_MAX*32)-1)
```

Comment:
I've had STL difficulty allocating vector<CellData> that high in the constructor unit test. I'm trying to work out why at the moment.